sdk/storage/Azure.Storage.DataMovement/src/Shared/DataMovementExtensions.cs

```diff
@@ -11,42 +11,36 @@ namespace Azure.Storage.DataMovement
 {
     internal static class DataMovementExtensions
     {
-        public static byte[] UriToByteArray(this StorageResource storageResource)
+        // string to byte array
+        public static byte[] ToByteArray(this string query, int? bufferSize = default)
         {
-            // Create Source Root
-            byte[] byteArray;
-            if (storageResource.CanProduceUri == ProduceUriType.ProducesUri)
-            {
-                Uri sourceUri = storageResource.Uri;
-                byteArray = string.Format(
-                    CultureInfo.InvariantCulture,
-                    "{0}{1}{2}",
-                    sourceUri.Authority,
-                    Uri.SchemeDelimiter,
-                    sourceUri.AbsolutePath).ToByteArray();
-            }
-            else
-            {
-                byteArray = storageResource.Path.ToByteArray();
-            }
-            return byteArray;
+            bufferSize ??= query.Length;
+
+            // Convert query to byte array.
+            byte[] arr = new byte[bufferSize.Value];
+            byte[] queryArr = Encoding.UTF8.GetBytes(query);
+            Array.Copy(queryArr, arr, queryArr.Length);
```

Comment:
Oh this is so that we have the padding of the null characters at the end.

But also I realized this lives in the wrong place since this is only used by our tests. I'm going to remove this.